tls/s2n_alerts.c

```diff
@@ -151,6 +151,20 @@ static bool s2n_handle_as_warning(struct s2n_connection *conn, uint8_t level, ui
     return type == S2N_TLS_ALERT_USER_CANCELED;
 }
 
+S2N_RESULT s2n_alerts_close_if_fatal(struct s2n_connection *conn, struct s2n_blob *alert)
+{
+    RESULT_ENSURE_REF(conn);
+    RESULT_ENSURE_REF(alert);
+    RESULT_ENSURE_EQ(alert->size, S2N_ALERT_LENGTH);
+    /* Only one alert should currently be treated as a warning */
+    if (alert->data[1] == S2N_TLS_ALERT_NO_RENEGOTIATION) {
+        RESULT_ENSURE_EQ(alert->data[0], S2N_TLS_ALERT_LEVEL_WARNING);
```

Comment:
Are alert identifiers unique to each alert level? Should this be:

```suggestion
    if (alert->data[0] ==  S2N_TLS_ALERT_LEVEL_WARNING && alert->data[1] == S2N_TLS_ALERT_NO_RENEGOTIATION) {
```

Reply:
They're unique regardless of level. TLS1.3 dumped the idea of levels altogether.

I prefer considering it an error if we encounter a non-warning no_renegotiation, since that should never happen.